### PR TITLE
[master] Add deprecation support

### LINKED ISSUE #21621: Deprecation should be supported for following constructs
**Description:**
$subject for 

- source annotation,
- source parameter
- source object field;

### LINKED ISSUE #21907: Fully qualified name of deprecated construct not given in the warning
Currently, only the name of the construct is given in the warnings for deprecated construct usage. It should ideally give the fully qualified name of the construct.

### LINKED ISSUE #22086: Code Analyser is not available for annotation Attachements
**Description:**
$subject

**Affected Versions:**
1.2.0

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

Fixes #21621 
Fixes #21907 
Fixes #22086 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/antlr4/BallerinaParserListener.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Generated from BallerinaParser.g4 by ANTLR 4.5.3
+// Generated from /home/kavindu/WSO2-GIT/test/ballerina-lang/compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4 by ANTLR 4.5.3
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/antlr4/BallerinaParserListener.java:1]: Please undo this change.

Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/antlr4/BallerinaParserListener.java]: Done

**Comment by MaryamZi** [comment]
Can we add tests for the code analysis validation of annotation attachments also?

e.g.,

```ballerina
type Foo record {|
    string s;
    int...;
|};

annotation Foo f on type;

Foo f = {s: "str", i: 1}; // invalid key 'i': identifiers cannot be used as rest field keys, expected a string literal or an expression

@f {
    s: "str",
    i: 1 // currently (1.2.0)  no error here but we should get the same error as above
}
type Bar object {
    
};
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationExpressionNegative.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationExpressionNegative.java:1]: Shall we call this `AnnotationExpressionCodeAnalysisNegative.java` and similarly the .bal file `annot_attachment_expression_code_analysis_negative.bal`?

Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationExpressionNegative.java]: Done

**Comment by KavinduZoysa** [comment]
> Can we add tests for the code analysis validation of annotation attachments also?
> 
> e.g.,
> 
> ```ballerina
> type Foo record {|
>     string s;
>     int...;
> |};
> 
> annotation Foo f on type;
> 
> Foo f = {s: "str", i: 1}; // invalid key 'i': identifiers cannot be used as rest field keys, expected a string literal or an expression
> 
> @f {
>     s: "str",
>     i: 1 // currently (1.2.0)  no error here but we should get the same error as above
> }
> type Bar object {
>     
> };
> ```

Added
